xml_converter/src/xml_converter.cpp

```diff
@@ -456,14 +454,14 @@ string create_burrito_data_folder() {
     const char* home_dir = getenv("HOME");
     if (home_dir == nullptr) {
         std::cerr << "Error: HOME environment variable is not set." << std::endl;
-        return "";
+        exit(0);
```

Comment:
Instead of `exit()`'ing you should throw an error. This way you can take advantage of the C++ error handling system on the off chance that you want to be able to "handle" this critical error without the entire program crashing, just the relevant part crashing. (`try` / `catch` / `throw`). In other languages this also enables stack traces but I cant remember how/if stack traces work in c++. `exit(0)` will exit the entire program regardless of anything.

Additionally `exit(0)` will exit with the return code `0`. Return code `0` is typically (but unfortunately not always in a few ancient core utility programs) the return code for "nothing went wrong". If you were to use `exit()` here it should have a number other than `0`, probably `1`. But that is irrelevant if you use `throw` instead.